src/expr/mod.rs

```diff
@@ -134,7 +134,20 @@ impl Function {
         }
     }
 
-    /// Returns the `DataType` of a column filterd by the current `Function`
+    /// Returns an equivalent `Function` for which the expressions
+    /// with values have been simplified
+    ///
+    /// Examples:
+    /// a > 3 * 5 -> a > 15
+    /// 3.5 - 1 = b -> 2.5 = b
+    pub fn simplify(&self) -> Self {
```

Comment:
Why do you need to simplify? Why not just look at the DataType and check it is a value ?